### PR TITLE
Fix encrypting + changes API  due to changes on backend

### PR DESCRIPTION
**Pull request by dpyskunova**

### CODE REVIEW

File: src/components/Common/Button/FbLoginButton/FbLoginButton.tsx [hunk 1]
```diff
@@ -26,17 +26,19 @@ export class FbLoginButton extends Component<IFacebookRegisterProps, {}> {
       location: (response.location && response.location.name) || ''
     };
 
-    auth(userDetails).then(response => {
+    auth('/user_register', userDetails).then(response => {
       const token = response.token;
```
Comment (fstodulski) [inline src/components/Common/Button/FbLoginButton/FbLoginButton.tsx]: u can also try
`const {token} = response;`

File: src/components/Login/Login.tsx [hunk 2]
```diff
@@ -54,15 +54,13 @@ export class Login extends React.Component <ILoginProps, ILoginState> {
   };
 
   authenticate = (userDetails:any) => {
-    auth(userDetails).then(response => {
-      if (response.sprededResponse.password && !(encrypt(response.sprededResponse.password, 10) === encrypt(userDetails.password, 10))) {
+    auth('/user_login', userDetails).then(response => {
         const token = response.token;
```
Comment (fstodulski) [inline src/components/Login/Login.tsx]: `const {token} = response;`

File: src/helpers/db.tsx [hunk 3]
```diff
@@ -1,9 +1,10 @@
 import { env } from '../env/environment';
 import {Fetch} from "./fetch";
 
-export function auth(userDetails: any) {
+export function auth(route:string, userDetails: any) {
+  console.log(env.publicRoutes.concat(route));
```
Comment (fstodulski) [inline src/helpers/db.tsx]: remove console logs before pushing

File: src/helpers/fetch.ts [hunk 4]
```diff
@@ -8,14 +8,21 @@ export abstract class Fetch {
     const securedHeader: RequestInit = {
       ...options,
       headers: {
-        Accept: 'application/json',
+        'Accept': 'application/json',
         'Content-Type': 'application/json',
         'token': token
       }
     };
```
Comment (fstodulski) [inline src/helpers/fetch.ts:16]: You will recive from server all data
200, 404, 500 for example, 
all you need is to use `.then()` and `.catch()`

**Review by fstodulski — CHANGES_REQUESTED** [review]

File: src/helpers/fetch.ts (hunk 4 above)
Comment (dpyskunova) [reply src/helpers/fetch.ts:16]: Please tell me in the room what did you mean here.

**Review by suezzo — APPROVED** [review]

File: src/components/Common/Button/FbLoginButton/FbLoginButton.tsx [hunk 5]
```diff
@@ -26,17 +26,19 @@ export class FbLoginButton extends Component<IFacebookRegisterProps, {}> {
...
     };
 
-    auth(userDetails).then(response => {
-      const token = response.token;
-      localStorage.setItem('token', token);
-      localStorage.setItem(
+    auth('/user_register', userDetails).then(response => {
+      const {token} = response;
+      token && localStorage.setItem('token', token);
+      response.sprededResponse && localStorage.setItem(
```
Comment (stanimirtt) [inline src/components/Common/Button/FbLoginButton/FbLoginButton.tsx]: typo -> spreaded

File: src/components/Login/Login.tsx [hunk 6]
```diff
@@ -54,15 +54,13 @@ export class Login extends React.Component <ILoginProps, ILoginState> {
...
-        const token = response.token;
-        localStorage.setItem('userDetails', JSON.stringify(response.sprededResponse));
-        localStorage.setItem('token', response.token);
-        this.props.stores.userDetails.user  =  {...response.sprededResponse, token};
-      } else {
-        console.log('The username and password did not match.');
-      }
+    auth('/user_login', userDetails).then(response => {
+        const {token} = response;
+        response.foundedUser && localStorage.setItem('userDetails', JSON.stringify(response.foundedUser));
```
Comment (stanimirtt) [inline src/components/Login/Login.tsx]: found, not founded.

File: src/components/Common/Button/FbLoginButton/FbLoginButton.tsx (hunk 5 above)
Comment (fstodulski) [reply src/components/Common/Button/FbLoginButton/FbLoginButton.tsx]: Fixed
`
 spreadedResponse,
          token,
`

**Review by anastasiialisikh — APPROVED** [review]

**Review by fstodulski — APPROVED** [review]

**Pull request merged**